src/unit_tests/agentlessd/test_agentlessd.c

```diff
@@ -0,0 +1,157 @@
+}
+
+#define HOST "host"
+#define SCRIPT "script"
+#define AGTTYPE "agttype"
+#define MSG "msg"
+#define STR_HELPER(s) #s
+#define STR(s) STR_HELPER(s)
+#define ADT 1667392718
+
+void test_gen_diff_alert(void **state) {
+    (void) state;
+
+    expect_string(__wrap_fopen, path, DIFF_DIR "/" HOST "->" SCRIPT "/diff." STR(ADT));
+    expect_string(__wrap_fopen, mode, "r");
+    will_return(__wrap_fopen, 1);
+
+    expect_fread(DIFF_DIR "/" HOST "->" SCRIPT "/diff." STR(ADT), BUFFER_SIZE);
+
+    expect_SendMSG_call("ossec: agentless: Change detected:\n" STR_MORE_CHANGES, 
+                "(" SCRIPT ") " HOST "->wazuh-agentlessd",
+                LOCALFILE_MQ, 0);
+
+    // save_agentless_entry() call
+    expect_string(__wrap_fopen, path, AGENTLESS_ENTRYDIR "/(" SCRIPT ") " HOST);
+    expect_string(__wrap_fopen, mode, "w");
+    will_return(__wrap_fopen, 2);
+
+    expect_any(__wrap_fprintf, __stream);
+    expect_string(__wrap_fprintf, formatted_msg, "type: diff\n");
+    will_return(__wrap_fprintf, 0);
+
+    expect_fclose((FILE *)2, 0);
+    // save_agentless_entry() call
+
+    expect_fclose((FILE *)1, 0);
+
+    gen_diff_alert(HOST, SCRIPT, ADT);
+}
+
+void test_send_log_msg_ok(void **state) {
+    (void) state;
+
+    expect_string(__wrap_SendMSG, message, MSG);
+    expect_string(__wrap_SendMSG, locmsg, "(" SCRIPT ") " HOST "->" SYSCHECK);
+    expect_any(__wrap_SendMSG, loc);
+    will_return(__wrap_SendMSG, 0);
+
+    send_log_msg(SCRIPT, HOST, MSG);
+}
+
+void test_send_intcheck_msg_ok(void **state) {
+    (void) state;
+
+    expect_string(__wrap_SendMSG, message, MSG);
+    expect_string(__wrap_SendMSG, locmsg, "(" SCRIPT ") " HOST "->" SYSCHECK);
+    expect_any(__wrap_SendMSG, loc);
+    will_return(__wrap_SendMSG, 0);
+
+    send_intcheck_msg(SCRIPT, HOST, MSG);
+}
+
+void test_save_agentless_entry_ok(void **state) {
+    (void) state;
+
+    expect_string(__wrap_fopen, path, AGENTLESS_ENTRYDIR "/(" SCRIPT ") " HOST);
+    expect_string(__wrap_fopen, mode, "w");
+    will_return(__wrap_fopen, 1);
+
+    expect_any(__wrap_fprintf, __stream);
+    expect_string(__wrap_fprintf, formatted_msg, "type: " AGTTYPE "\n");
+    will_return(__wrap_fprintf, 0);
+
+    expect_any(__wrap_fclose, _File);
+    will_return(__wrap_fclose, 0);
+
+    save_agentless_entry(HOST, SCRIPT, AGTTYPE);
+}
+
+void test_Agentlessd_ok(void **state) {
+    (void) state;
+
+    os_malloc(2 * sizeof(agentlessd_entries *), lessdc.entries);
+    os_calloc(1, sizeof(agentlessd_entries), lessdc.entries[0]);
+    lessdc.entries[1] = NULL;
+
+    lessdc.entries[0]->server = NULL;
+    lessdc.entries[0]->command = NULL;
+    lessdc.entries[0]->options = "";
+    lessdc.entries[0]->type = NULL;
+    lessdc.entries[0]->frequency = 86400;
+    lessdc.entries[0]->state = 0;
+    lessdc.entries[0]->current_state = 0;
+    lessdc.entries[0]->port = 0;
+    lessdc.entries[0]->error_flag = 0;
+
+    expect_StartMQ_call(DEFAULTQUEUE, WRITE, 0);
+
+    Agentlessd();
```

Comment:
Just a question: can we confirm that the line `494,17:                 run_periodic_cmd(lessdc.entries[i], test_it);` isn't executed during the tests?

Reply:
I don't think so, but i'll control it